mlflow/types/utils.py

```diff
@@ -724,17 +725,52 @@ def _validate_dict_examples(examples, num_items=None):
         _validate_keys_match(example, first_keys)
 
 
+def _is_pep585_supported() -> bool:
+    """
+    Is https://peps.python.org/pep-0585 supported in the current Python version?
+    """
+    return sys.version_info[:2] >= (3, 9)
+
+
+def _is_list_str(type_hint: Any) -> bool:
+    if type_hint == List[str]:
+        return True
+
+    if _is_pep585_supported():
+        try:
+            return type_hint == list[str]
+        except TypeError:
```

Comment:
Is `TypeError` the only possible error here? Maybe we always return False for any exception?

Reply:
Good suggestion. Changed the error to `Exception`.